src/workflows-app/RunDetails.js

```diff
@@ -151,6 +157,33 @@ export const BaseRunDetails = (
     [signal, workspaceId]
   );
 
+  const inProgressElement = (workflow) => {
+    if (workflow.status === 'Running') {
+      return span({ style: { fontStyle: 'italic' } }, ['In progress - ']);
+    }
+  };
+
+  const calculateTotalCost = (callObjects) => {
+    let total = 0;
+    Object.values(callObjects).forEach((call) => {
+      if (!call[0].taskStartTime) {
+        total += 0;
+      } else if (call[0].taskEndTime) {
+        total += getTaskCost(call[0].vmCostUsd, call[0].taskStartTime, call[0].taskEndTime);
+      } else {
+        total += getTaskCost(call[0].vmCostUsd, call[0].taskStartTime);
+      }
+    });
+    return total;
+  };
+
+  useEffect(() => {
+    if (callObjects) {
+      const cost = calculateTotalCost(callObjects);
+      setTaskCostTotal(cost);
+    }
+  }, [callObjects]);
```

Comment:
It looks like `useMemo` might be preferable over `useEffect` here.

https://react.dev/reference/react/useMemo

```suggestion
  const taskCostTotal = useMemo(() => {
    return callObjects ? calculateTotalCost(callObjects) : undefined;
  }, [callObjects]);
```